### PR TITLE
stake-contract: remove unused `check_new_block`

### PR DESCRIPTION
**Pull request by herr-seppia**

### CODE REVIEW

File: rusk/src/lib/node/rusk.rs [hunk 1]
```diff
@@ -465,7 +465,32 @@ impl Rusk {
...
+    /// Opens a session for query, setting a block height of zero since this
+    /// doesn't affect the result.
+    pub(crate) fn query_session(
+        &self,
+        commit: Option<[u8; 32]>,
+    ) -> Result<Session> {
+        self._session(0, commit)
+    }
+
+    pub(crate) fn _session(
```
Comment (fed-franz) [inline rusk/src/lib/node/rusk.rs:493]: Is the trailing `_` necessary?

**Review by fed-franz — DISMISSED** [review]
LGTM

File: rusk/src/lib/node/rusk.rs (hunk 1 above)
Comment (herr-seppia) [reply rusk/src/lib/node/rusk.rs:493]: It's a common way to let the dev know to not directly call the function (neither locally)

Comment (herr-seppia) [reply rusk/src/lib/node/rusk.rs]: Btw, I removed the `pub(crate)` accessor and added a comment to warn users to not call it directly

**Review by fed-franz — APPROVED** [review]

**Pull request merged**